cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -412,38 +390,89 @@ def test_cloud_provision_from_template_with_attached_disks(
+            {'boot_index': 0 if i == 0 else -1,
+            'uuid': volume,
+            'device_name': device_name.format(chr(ord("a") + i))})
+
+        if i == 0:
+            provider.mgmt.capi.volumes.set_bootable(volume, True)
+
+    method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
+
+    view = navigate_to(method, 'Details')
+    former_method_script = view.script.get_value()
+
+    disk_mapping = []
+    for mapping in device_mapping:
+        one_field = dedent("""{{
+            :boot_index => {boot_index},
+            :uuid => "{uuid}",
+            :device_name => "{device_name}",
+            :source_type => "volume",
+            :destination_type => "volume",
+            :volume_size => 1,
+            :delete_on_termination => false
+        }}""")
+        disk_mapping.append(one_field.format(**mapping))
+
+    volume_method = dedent("""
+        clone_options = {{
+        :image_ref => nil,
+        :block_device_mapping_v2 => [
+            {}
+        ]
+        }}
+
+        prov = $evm.root["miq_provision"]
+        prov.set_option(:clone_options, clone_options)
+    """)
+    with update(method):
+        method.script = volume_method.format(",\n".join(disk_mapping))
 
+    @request.addfinalizer
+    def _finish_method():
         with update(method):
-            disk_mapping = []
-            for mapping in device_mapping:
-                disk_mapping.append(ONE_FIELD.format(*mapping))
-            method.script = VOLUME_METHOD.format(", ".join(disk_mapping))
-
-        def _finish_method():
-            with update(method):
-                method.script = """prov = $evm.root["miq_provision"]"""
-        request.addfinalizer(_finish_method)
-
-        instance = appliance.collections.cloud_instances.create(vm_name,
-                                                                provider,
-                                                                form_values=inst_args)
-
-        for volume_id in volumes:
-            soft_assert(vm_name in provider.mgmt.volume_attachments(volume_id))
-        for volume, device in device_mapping:
-            soft_assert(provider.mgmt.volume_attachments(volume)[vm_name] == device)
-        instance.mgmt.delete()  # To make it possible to delete the volume
+            method.script = former_method_script
+
+    instance = appliance.collections.cloud_instances.create(vm_name,
+                                                            provider,
+                                                            form_values=inst_args)
+
+    @request.addfinalizer
+    def delete_vm_and_wait_for_gone():
+        instance.mgmt.delete()
         wait_for(lambda: not instance.exists_on_provider, num_sec=180, delay=5)
 
+    for volume_id in volumes:
+        attachments = provider.mgmt.volume_attachments(volume_id)
+        soft_assert(
+            vm_name in attachments,
+            'The vm {} not found among the attachemnts of volume {}:'.format(
```

Comment:
:+1: Nice, did this syntax help with tracebacks on any test failures you ran into?

Reply:
It might, but I didn't test it. These tests are already passing and I didn't try to introduce any failure.